sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -496,21 +507,41 @@ const pi_uint32 DynamicBatchStartSize = 4;
 
 struct _pi_queue : _pi_object {
   _pi_queue(ze_command_queue_handle_t Queue,
-            ze_command_queue_handle_t CopyQueue, pi_context Context,
-            pi_device Device, pi_uint32 BatchSize, bool OwnZeCommandQueue,
-            pi_queue_properties PiQueueProperties = 0)
+            std::vector<ze_command_queue_handle_t> &CopyQueues,
+            pi_context Context, pi_device Device, pi_uint32 BatchSize,
+            bool OwnZeCommandQueue, pi_queue_properties PiQueueProperties = 0)
       : ZeComputeCommandQueue{Queue},
-        ZeCopyCommandQueue{CopyQueue}, Context{Context}, Device{Device},
+        ZeCopyCommandQueues{CopyQueues}, Context{Context}, Device{Device},
         QueueBatchSize{BatchSize > 0 ? BatchSize : DynamicBatchStartSize},
         OwnZeCommandQueue{OwnZeCommandQueue}, UseDynamicBatching{BatchSize ==
                                                                  0},
         PiQueueProperties(PiQueueProperties) {}
 
   // Level Zero compute command queue handle.
   ze_command_queue_handle_t ZeComputeCommandQueue;
-  // Level Zero copy command command queue handle. This might not be available
-  // depending on user preference and/or target device.
-  ze_command_queue_handle_t ZeCopyCommandQueue;
+  // Vector of Level Zero copy command command queue handles.
+  // Some (or all) of these handles may not be available depending on user
+  // preference and/or target device.
+  // In this vector, link copy engines, if available, come first followed by
+  // main copy engine, if available.
+  std::vector<ze_command_queue_handle_t> ZeCopyCommandQueues;
+
+  // One of the many available copy command queues will be used for
+  // submitting command lists to. This variable stores index of the last used
+  // copy command queue in the ZeCopyCommandQueues vector.
+  int32_t LastUsedCopyCommandQueueIndex = -1;
+
+  // This function will return one of possibly multiple available copy queues.
+  // Currently, a round robin strategy is used.
+  // It will return nullptr if no copy command queues are available for use.
+  ze_command_queue_handle_t
+  getZeCopyCommandQueue(int &CopyQueueIndex,
```

Comment:
I suggest to make first argument a pointer for consistency

Reply:
Addressed this in the next patch. Thanks